NMS-12629: Trapd is missing in the docs (#5305)
OpenNMS/opennms: docs/modules/reference/pages/daemons/daemon-config-files/trapd.adoc

= Trapd

Handles processing of xref:operation:events/sources/snmp-traps.adoc[SNMP traps] into events.
Based around traditional UDP-based SNMP traps, it receives traps remote devices create.

> It enables OpenNMS to receive SNMP traps and turn them into events, and then into alarms and notifications, much as xref:reference:daemons/daemon-config-files/syslogd.adoc[syslogd] does for syslog entries.

Reviewer: ```suggestion
It enables OpenNMS to receive SNMP traps and convert them into events, and convert those events into alarms and notifications.
Its functionality is similar to how xref:reference:daemons/daemon-config-files/syslogd.adoc[syslogd] handles syslog entries.
```

= Trapd

> Handles processing of xref:operation:events/sources/snmp-traps.adoc[SNMP traps] into events.

Author: When we updated the daemon section, we wanted to eliminate the verbiage for all the daemons. (Originally they were "The trapd daemon provides ..." or "The syslog daemon enables...". We can revisit this in light of the style guide, but for now, let's keep it as is so that all the daemon chapters have the same structure.

> Handles processing of xref:operation:events/sources/snmp-traps.adoc[SNMP traps] into events.

Reviewer: Sounds good! I couldn't remember exactly what we'd decided, so I'll make a note of this.

> Handles processing of xref:operation:events/sources/snmp-traps.adoc[SNMP traps] into events.

Author: No problem. I think we updated the daemon section before you started ...

> Based around traditional UDP-based SNMP traps, it receives traps remote devices create.

Reviewer: ```suggestion
Based around traditional UDP-based SNMP traps, it receives traps created by remote devices.
```

> It enables OpenNMS to receive SNMP traps and convert them into events, and convert those events into alarms and notifications.

Reviewer 2: ```suggestion
It enables OpenNMS to receive SNMP traps and convert them into events, and, if configured accordingly, convert those events into alarms and notifications.
```

Maybe something like this? Since alarming and notifying are not configured by default. Not sure about the commas in English :see_no_evil:

> It enables OpenNMS to receive SNMP traps and convert them into events, and, if configured accordingly, convert those events into alarms and notifications.

Reviewer: ```suggestion
It enables OpenNMS to receive SNMP traps and convert them into events and, if configured accordingly, convert those events into alarms and notifications.
```

Its functionality is similar to how xref:reference:daemons/daemon-config-files/syslogd.adoc[syslogd] handles syslog entries.

Trapd supports V1 traps, V2 traps and notifications, and V3 traps and notifications.

> Trapd accepts traps with any community name (V1, V2c).

Reviewer: ```suggestion
It also accepts traps with any community name (V1, V2c).
```

You can use that community name in event definitions as part of a mask.

== How it works

When an SNMP trap arrives, trapd parses the message, comparing the sender to known nodes.

> If the sender is an unknown node and the `new-suspect-on-trap` parameter is set to "true", {page-component-title} sends an event to discover the node.

Reviewer 2: ```suggestion
If the sender is a not monitored node and the `new-suspect-on-trap` parameter is set to "true", {page-component-title} sends an event to [discover](https://docs.opennms.com/horizon/30/operation/provisioning/auto-discovery.html) the node.
```

> If the sender is not a monitored node and the `new-suspect-on-trap` parameter is set to "true", {page-component-title} sends an event to xref:operation:provisioning/auto-discovery.adoc[discover] the node.

Author: ```suggestion
If the sender is not a monitored node and the `new-suspect-on-trap` parameter is set to `true`, {page-component-title} sends an event to xref:operation:provisioning/auto-discovery.adoc[discover] the node.
```

> If the sender is not a monitored node and the `new-suspect-on-trap` parameter is set to "true", {page-component-title} sends an event to xref:operation:provisioning/auto-discovery.adoc [discover] the node.

Reviewer: ```suggestion
If the sender is not a monitored node and the `new-suspect-on-trap` parameter is set to `true`, {page-component-title} sends an event to xref:operation:provisioning/auto-discovery.adoc [discover] the node.
```

> Otherwise, {page-component-title} tags the trap to the correct node, broadcasts it to eventd, and the trap becomes a {page-component-title} event.

Reviewer 2: ```suggestion
Otherwise, {page-component-title} tags the trap to the monitored node, broadcasts it to eventd, and the trap becomes a {page-component-title} event.
```

| trapd.log
| Logs relating to trapd server events.
|===

== Additional information

> For information on how to automate alarms and reduce the number of messages, see the syslogd example in https://opennms.discourse.group/t/syslogd-automations/1454[Syslogd Automations] on Discourse.

Reviewer: ```suggestion
For information on how to automate alarms and reduce the number of messages, see https://opennms.discourse.group/t/syslogd-automations/1454[Syslogd Automations] on Discourse.
```

You can configure the eventd subsystem to discard unwanted SNMP traps.

> To do so, use the logmsg `discardtraps`.

Reviewer: ```suggestion
To do so, use the `discardtraps` logmsg.
```

> See xref:operation:events/event-definition.adoc#ga-events-anatomy-of-an-event[Anatomy of an event] in the Event Daemon Configuration section.

Reviewer: ```suggestion
See xref:operation:events/event-definition.adoc#ga-events-anatomy-of-an-event[Anatomy of an event] in the Event Daemon Configuration section for more information.
```
